server/cluster/cluster.go

```diff
@@ -1349,6 +1349,53 @@ func (c *RaftCluster) SlowStoreRecovered(storeID uint64) {
 	c.core.SlowStoreRecovered(storeID)
 }
 
+// NeedAwakenAllRegionsInStore checks whether we should do AwakenRegions operation.
+func (c *RaftCluster) NeedAwakenAllRegionsInStore(storeID uint64) (needAwaken bool, slowStoreIDs []uint64) {
```

Comment:
There's no point to have needAwaken as slowStoreIDs itself is sufficient to indicate that (when it's empty or nil).

Reply:
There's a problem in this mechanism. So basically when there's a slow score store, it will ask other stores to awake hibernate regions every 30s.  Even if the slow score store don't have leaders anymore as they're transferred by the awaken hibernate regions. This is a problem because it will repeatedly awaken regions that are not necessary to awaken. 
I think the simple mitigate is to increase the interval from 30s to a reasonable large interval such as 10 minutes.

Reply:
> There's no point to have needAwaken as slowStoreIDs itself is sufficient to indicate that (when it's empty or nil).

IMO，`NeedAwakenAllRegionsInStore`'s naming contains that we should return a "state" whether we should do AwakenRegions, and that is why we use `needAwaken`.

Reply:
> There's a problem in this mechanism. So basically when there's a slow score store, it will ask other stores to awake hibernate regions every 30s. Even if the slow score store don't have leaders anymore as they're transferred by the awaken hibernate regions. This is a problem because it will repeatedly awaken regions that are not necessary to awaken. I think the simple mitigate is to increase the interval from 30s to a reasonable large interval such as 10 minutes.

IMP, too large interval is not a recommendable choice, which might lost in abnormal scenarios, i.e. Net Error. And TiKV side will automatically ignore redundant AwakenRegions requests if its regions are awaken by a previous AwakenRegions message.